homeassistant/components/viam/__init__.py

```diff
@@ -0,0 +1,366 @@
+        if isinstance(image, Image.Image):
+            image_bytes = image.tobytes()
+        if isinstance(image, RawImage):
+            image_bytes = image.data
+
+        image_string = base64.b64encode(image_bytes).decode()
+        return f"data:image/jpeg;base64,{image_string}"
+
+    async def _get_image(self, filepath: str | None, camera_entity: str | None):
+        if filepath is not None:
+            return await self.hass.async_add_executor_job(_fetch_image, filepath)
+        if camera_entity is not None:
+            image = await camera.async_get_image(self.hass, camera_entity)
+            return RawImage(image.content, image.content_type)
+
+        return None
+
+    async def _get_robot_client(
+        self, robot_secret: str | None, robot_address: str | None
+    ) -> RobotClient:
+        """Check initialized data to create robot client."""
+        address = self.data.get(CONF_ADDRESS)
+        payload = self.data.get(CONF_SECRET)
+        if self.data[CONF_CREDENTIAL_TYPE] == CRED_TYPE_API_KEY:
+            if robot_secret is None or robot_address is None:
+                raise ServiceValidationError(
+                    "The robot location and secret are required for this connection type.",
+                    translation_domain=DOMAIN,
+                    translation_key="robot_credentials_required",
+                )
+            address = robot_address
+            payload = robot_secret
+
+        if address is None or payload is None:
+            raise ServiceValidationError(
+                "The necessary credentials for the RobotClient could not be found.",
+                translation_domain=DOMAIN,
+                translation_key="robot_credentials_not_found",
+            )
+
+        credentials = Credentials(type=CRED_TYPE_LOCATION_SECRET, payload=payload)
+        robot_options = RobotClient.Options(
+            refresh_interval=0, dial_options=DialOptions(credentials=credentials)
+        )
+        return await RobotClient.at_address(address, robot_options)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Viam component."""
+    config_entries = hass.config_entries.async_entries(DOMAIN)
+    if not config_entries:
+        raise HomeAssistantError(
+            "No Viam config entries found",
+            translation_domain=DOMAIN,
+            translation_key="entry_not_found",
+        )
+
+    for config_entry in config_entries:
+        if config_entry.state != ConfigEntryState.LOADED:
+            raise HomeAssistantError(
+                f"{config_entry.title} is not loaded",
+                translation_domain=DOMAIN,
+                translation_key="entry_not_loaded",
+                translation_placeholders={
+                    "config_entry_title": config_entry.title,
+                },
+            )
+        manager: ViamManager = hass.data[DOMAIN][config_entry.entry_id]
+        manager.register_services()
+        break
+
```

Comment:
```suggestion
    register_services(hass)
```

`register_services` should be outside of the manager and when called the user should select the config entry for it.
Example:
-Get the manager similar to https://github.com/HipsterBrown/home-assistant-core/blob/b24581bf28bf4cdc7b90b36957a85e690fda6f56/homeassistant/components/easyenergy/services.py#L88
- Function should look like https://github.com/HipsterBrown/home-assistant-core/blob/b24581bf28bf4cdc7b90b36957a85e690fda6f56/homeassistant/components/easyenergy/__init__.py#L19